homeassistant/components/velux/__init__.py

```diff
@@ -108,10 +108,14 @@ class VeluxEntity(Entity):
 
     _attr_should_poll = False
 
-    def __init__(self, node: Node) -> None:
+    def __init__(self, node: Node, config_entry_id: str) -> None:
         """Initialize the Velux device."""
         self.node = node
-        self._attr_unique_id = node.serial_number
+        self._attr_unique_id = (
+            node.serial_number
+            if node.serial_number
```

Comment:
Is there a possibility that the node has today no serial number but, after a restart or whatever, has a serial number?
Or are we 100% sure that if a node has no serial number, it will never have a serial number

Reply:
I can't tell you exactly, but in the custom component of pawlizio (https://github.com/pawlizio/my_velux) this has been handled this way for a long time. Since no errors occur there, I would say that the devices will not get a serial number.